vulcan/lib/server/workflows/scripts/calc_umap.py

```diff
@@ -1,10 +1,26 @@
-from archimedes.functions.dataflow import output_path, input_path, input_var
+from archimedes.functions.dataflow import output_path, output_json, input_path, input_var
 from archimedes.functions.scanpy import scanpy as sc
 
 scdata = sc.read(input_path('nn_anndata.h5ad'))
+min_dist = float(input_var('min_dist'))
+spread = float(input_var('spread'))
+num_iters = int(input_var('num_iters'))
 
 # umap
-sc.tl.umap(scdata)
+sc.tl.umap(scdata, min_dist = min_dist, spread = spread, maxiter = num_iters)
+
+# color options
+color_options = {
+    'Cluster': None,
+    'Experiment': None,
+    'Tissue': None,
+    'Pool': None,
+    'Biospecimen Group': None,
+    'Gene': dict([ [ gene_id, None ] for gene_id in scdata.var_names ])
```

Comment:
This uses the list of reduced genes in the final UMAP for coloring - since the original matrix is stripped down this is the best we can do here, but some important genes might not be visualized this way (e.g. cd3e didn't show up in my test) - does this seem reasonable @dtm2451?

Reply:
Oh right this is my fault.  There is definitely a way to have the non-reduced data stick around.  It's always seemed a bit clunky to me, but I can fix that in the scripts after this gets merged!